app/controllers/register/controller.js

```diff
@@ -0,0 +1,33 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import zxcvbn from 'zxcvbn';
+
+export default class Register extends Controller {
+  @tracked passwordStrength = '';
+  @tracked disabledButton = true;
+  @tracked showPassword = false;
+
+  @action
+  setPasswordText(str) {
+    this.set('passwordText', str);
+  }
+
+  @action
+  valChange() {
+    const currentPassword = this.get('passwordText');
+    const { score: strengthScore } = zxcvbn(currentPassword);
+    if (strengthScore >= 2) {
+      this.disabledButton = false;
+    } else {
+      this.disabledButton = true;
+    }
```

Comment:
Setting values directly can lead to problems later. Think about what can possibly go wrong and how you can refactor this

Reply:
I think I will change the disabledButton to computed property like this:

```
export default class Register extends Controller {
  @tracked passwordStrength = '';
  @tracked strengthScore = 0;
  @tracked showPassword = false;

  get disabledButton() {
    if (this.strengthScore >= 2) {
      return false;
    } else {
      return true;
    }
  }
  
  @action
  valChange() {
    const currentPassword = this.get('passwordText');
    const { score } = zxcvbn(currentPassword);
    this.passwordStrength = score;
    this.strengthScore = score;
  }

  @action
  setPasswordText(str) {
    this.set('passwordText', str);
  }


  @action
  showPasswordClick() {
    this.showPassword = !this.showPassword;
  }
}
```